src/mixins/canvas_serialization.mixin.js

```diff
@@ -30,42 +46,48 @@ fabric.util.object.extend(fabric.StaticCanvas.prototype, /** @lends fabric.Stati
       : Object.assign({}, json);
 
     var _this = this,
-        renderOnAddRemove = this.renderOnAddRemove;
+        renderOnAddRemove = this.renderOnAddRemove,
+        abortController = new AbortController();
```

Comment:
You mean here?

Reply:
IMO loadFromJSON is a strange thing.
It should behave like loadSVGFromXXX and let the dev add objects to canvas.
In such a case it makes sense to pass an abort controller to the loading task, and for that task to be separated from canvas state.

Reply:
?

Reply:
that is an entire different issue.
LoadFromJSON today restore a previous canvas, is not a bulk loader, while it could totaly be, but we have enliveObjects from that.

What i mean is that we should not wrap the abortController functionality with custom code.
We should let it pass down by options, arrive to the only thing that can be interrupted and is the request of an image and let the developer call it.

When a developer call set('src', '....') he will verify if there is a previous abort controller going on and can cancel it or not.

Reply:
How can a dev use `set('src', '....')` with an abort controller?
I don't see how it's possible.
Meaning it will make one of the possibilities die out.
It is possible using `setSrc` but it's messy to tell the dev "if you want to set `src` with an abort controller then don't use `set`"
Regarding the json task... I think all async tasks should be abortable.

Reply:
> LoadFromJSON today restore a previous canvas, is not a bulk loader, while it could totaly be, but we have enliveObjects for that.

OK but what if the dev calls loadFromJSON before the prev task completed. And let's say the the prev task is really heavy. If we don't abort it it will override the incoming task once it completes. That's why I've handled in such a way and that's why I think that mixing canvas state and loading tasks is bad practice.